core/scripts/liquidity-mining/CalculateBalancerLPProviders.js

```diff
@@ -169,22 +146,88 @@ async function _updatePayoutAtBlock(bPool, blockNumber, shareHolderPayout, umaPe
 function _saveShareHolderPayout(shareHolderPayout, week) {
   // First, clean the shareHolderPayout of all zero recipients and convert from wei scaled number.
   for (shareHolder of Object.keys(shareHolderPayout)) {
-    if (shareHolderPayout[shareHolder] == "0") delete shareHolderPayout[shareHolder];
+    if (shareHolderPayout[shareHolder].toString() == "0") delete shareHolderPayout[shareHolder];
     else shareHolderPayout[shareHolder] = fromWei(shareHolderPayout[shareHolder]);
   }
 
   const savePath = `${path.resolve(__dirname)}/weekly-payouts/${week}_week_UMAsToDistribute.json`;
-  fs.writeFile(savePath, JSON.stringify(shareHolderPayout), err => {
-    if (err) return console.error(err);
-    console.log("🗄  File successfully written to", savePath);
+  fs.writeFileSync(savePath, JSON.stringify(shareHolderPayout));
+  console.log("🗄  File successfully written to", savePath);
+}
+
+// Find information about a given balancer pool `shares` returns a list of all historic LP providers. First tries to
+// query the pool directly based off the pool address. If this does not work then gets the addresses of the tokens
+// the pool holds and uses this to query the graph.
+async function _fetchBalancerPoolInfo(poolAddress) {
+  const SUBGRAPH_URL = process.env.SUBGRAPH_URL || "https://api.thegraph.com/subgraphs/name/balancer-labs/balancer";
+  const poolAddressQuery = `
+        {
+          pools (where: {id: "${poolAddress}"}) {
+            id
+            shares (first: 1000) {
+              userAddress {
+                id
+              }
+            }
+          }
+        }
+    `;
+
+  const poolAddressQueryResponse = await fetch(SUBGRAPH_URL, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json"
+    },
+    body: JSON.stringify({ query: poolAddressQuery })
   });
+
+  const poolAddressQueryData = (await poolAddressQueryResponse.json()).data;
+  if (poolAddressQueryData.pools.length > 0) {
```

Comment:
however, if the query returns blank, then we try and find the tokens held by the pool.